mlflow/utils/rest_utils.py

```diff
@@ -58,45 +106,21 @@ def http_request(
     if host_creds.client_cert_path is not None:
         kwargs["cert"] = host_creds.client_cert_path
 
-    def request_with_ratelimit_retries(max_rate_limit_interval, **kwargs):
-        response = requests.request(**kwargs)
-        time_left = max_rate_limit_interval
-        sleep = 1
-        while response.status_code == 429 and time_left > 0:
-            _logger.warning(
-                "API request to {path} returned status code 429 (Rate limit exceeded). "
-                "Retrying in %d seconds. "
-                "Will continue to retry 429s for up to %d seconds.",
-                sleep,
-                time_left,
-            )
```

Comment:
It was kind of nice to have this warning behavior here previously so that users know why their workflows are taking so long or eventually failing. Is there a way to log failures using `HTTPAdapter`?

Reply:
Not that I'm aware of. If the request fails ultimately owing to rate limit, the user can still see it, but I agree that the visibility on rate limit is lower in this case.